storage/rocksdb/ha_rocksdb.cc

```diff
@@ -11900,8 +12411,15 @@ int ha_rocksdb::external_lock(THD *const thd, int lock_type) {
       }
     }
     tx->m_n_mysql_tables_in_use++;
-    rocksdb_register_tx(rocksdb_hton, thd, tx);
+    rocksdb_register_tx(rocksdb_hton, thd, tx, (lock_type == F_WRLCK));
```

Comment:
This seems like it should be `lock_type == F_WRLCK || lock_type == F_RDLCK` so that `select lock in share mode` can also read the latest data?

Reply:
Agree that `select ... lock in share mode` should read the latest data.
However, `F_RDLCK` is passed for both SELECTs and `SELECT ... LOCK IN SHARE MODE`. 
Changed the check to analyze `m_lock_rows`.